app/models/clubhouse.rb

```diff
@@ -0,0 +1,37 @@
+class Clubhouse < ApplicationRecord
+  has_many :posts, dependent: :destroy
+  has_many :memberships, dependent: :destroy
+
+  validates :title, presence: true
+
+  def self.create_with_owner(owner, properties)
+    clubhouse = Clubhouse.create(properties)
+    membership = Membership.new(
+      user: owner,
+      clubhouse:,
+      role: Membership::OWNER_ROLE
+    )
+    return clubhouse if membership.save
+
+    nil
+  end
+
+  def owners
+    users_of_role(Membership::OWNER_ROLE)
+  end
+
+  def admins
+    users_of_role(Membership::ADMIN_ROLE)
+  end
+
+  def members
+    users_of_role(Membership::MEMBER_ROLE)
+  end
+
+  private
+
+  def users_of_role(role)
+    role_memberships = memberships.where(role:)
+    User.where(id: role_memberships.select(:user_id))
```

Comment:
There is no need to filter over all users from the users table, but only over the users that are associated to the clubhouse instance. I think you should add the `has_many :users, through: :memberships` association and use it here
```suggestion
    users.where(id: role_memberships.select(:user_id))
```

Reply:
Right, done!

Reply:
Right, done!